iac-extensions/terraform/src/main/java/org/sonar/iac/terraform/checks/DisabledS3EncryptionCheck.java

```diff
@@ -29,8 +29,12 @@
   private static final String MESSAGE = "Make sure not using server-side encryption is safe here.";
 
   @Override
-  protected void checkResource(CheckContext ctx, BlockTree resource) {
-    if (isS3Bucket(resource) && PropertyUtils.isMissing(resource, "server_side_encryption_configuration")) {
+  protected void registerChecks() {
+    register(DisabledS3EncryptionCheck::checkBucket, "aws_s3_bucket");
+  }
+
+  private static void checkBucket(CheckContext ctx, BlockTree resource) {
+    if (PropertyUtils.isMissing(resource, "server_side_encryption_configuration")) {
```

Comment:
Just an idea: when the check is so short, it could be easier to use a lambda:
```
    register((ctx, resource) -> {
      if (PropertyUtils.isMissing(resource, "server_side_encryption_configuration")) {
        reportResource(ctx, resource, MESSAGE);
      }
    }, "aws_s3_bucket");
```
However, it could be annoying to have the name of the resource as the last argument.